services/121-service/test/registrations/pagination/send-message.test.ts

```diff
@@ -228,4 +228,59 @@ describe('send arbitrary messages to set of registrations', () => {
       TwilioErrorCodes.toNumberDoesNotExist,
     );
   });
+
+  it('should not send message twice to the same registrations on fast consecutive requests', async () => {
+    // Arrange
+    const message = 'Send this message five times';
+    const finalMessage =
+      'Send this message at the end to check if all are complete, so we know we need to stop waiting';
+
+    const amountOfSends = 5;
+
+    // Act: Send the same message in fast consecutive requests
+
+    for (let i = 0; i < amountOfSends; i++) {
+      await sendMessage(
+        accessToken,
+        programIdOCW,
+        [registrationOCW1.referenceId],
+        message,
+      );
+    }
+
+    // Send a final message to know when to stop waiting
+    await sendMessage(
+      accessToken,
+      programIdOCW,
+      [registrationOCW1.referenceId],
+      finalMessage,
+    );
+
+    // Wait for all messages to complete
+    await waitForMessagesToComplete({
+      programId: programIdOCW,
+      referenceIds: [registrationOCW1.referenceId],
+      accessToken,
+      expectedMessageAttribute: {
+        key: 'body',
+        values: [finalMessage],
+      },
+    });
+
+    // Assert
+    const messageHistory = (
+      await getMessageHistory(
+        programIdOCW,
+        registrationOCW1.referenceId,
+        accessToken,
+      )
+    ).body;
+
+    // Filter message history to only include messages with body equal to message
+    const filteredMessageHistory = messageHistory.filter(
+      (msg) => msg.attributes.body === message,
+    );
+
+    expect(filteredMessageHistory.length).toBe(amountOfSends);
```

Comment:
so the assertion of 5 === 5, here means that each is sent only once right? Is the 5 chosen here in light of statistics? We could theoretically have functionally demonstrated this also with 1 message (+ the final). But the idea is that with 5 messages all being sent only once, this is more convincing?

I don't find the comments/etc super-intuitive to understand what's happening.